modules/architecture-machine-config-pools.adoc

```diff
@@ -0,0 +1,24 @@
+// Module included in the following assemblies:
+//
+// * architecture/control-plane.adoc
+
+[id="architecture-machine-config-pools_{context}"]
+= Node configuration management with machine config pools
+
+Machines that manage the cluster and its workloads are divided into groups based on the types of resources they handle. These groups of machines are called machine config pools (MCP). Each MCP manages a set of nodes and its corresponding machine configs. The role of the node determines which MCP it belongs to; the MCP governs nodes based on its assigned node role label. Nodes in an MCP have the same configuration; this means nodes can be scaled up and torn down in response to increased or decreased workloads.
+
+By default, there are two MCPs created by the cluster when it is installed: `master` and `worker`. Each default MCP has a defined configuration applied by the Machine Config Operator (MCO), which is responsible for managing MCPs and facilitating MCP upgrades. You can create additional MCPs, or custom pools, to manage nodes that have custom use cases that extend outside of the default node types.
+
+[NOTE]
+====
+A node can only be included in one machine config pool.
+====
+
+Custom pools are pools that inherit their configurations from the worker pool. They use any machine config targeted for the worker pool, but add the ability to deploy changes only targeted at the custom pool. Since a custom pool inherits its configuration from the worker pool, any change to the worker pool is applied to the custom pool as well. Custom pools inheriting configurations from master nodes are not supported by the MCO.
+
+[NOTE]
+====
+It is recommended to have a custom pool for every node role you want to manage in your cluster. For example, if you create infra nodes to handle infra workloads, it is good practice to create a custom infra MCP to group those nodes together. If you apply an `infra` role label to a worker node so it has the `worker,infra` dual label, it is still considered a worker node by the MCO because it is managed by the worker MCP. If you strip the `worker` label from a node and apply the `infra` label without grouping it in a custom pool, the node is not recognized by the MCO and is unmanaged by the cluster.
```

Comment:
`If you apply an `infra` role label to a worker node so it has the `worker,infra` dual label,` -> `If you apply an `infra` role label to a worker node so it has the `worker,infra` dual label, but do not have a custom infra MCO, the MCO considers it a worker node...` to clarify

Reply:
@yuqi-zhang There have been several doc BZs reported by customers with confusion on whether nodes labeled with `worker,infra` will be charged for running infra workloads only. From the infra call yesterday, it sounded like the future of infra nodes is to maintain the dual `worker,infra` label, so I think a clear statement somewhere in this section would be beneficial.

So a quick question for clarity:

*If a node is labeled with `worker,infra` without a custom infra MCP, would the workloads on that node require a subscription since it's managed by the worker MCP?*

I'll write an explicit statement for this after I'm clear on what we want to promote.

Reply:
I think so long as its marked as an infra node, and we taint it such that user workloads do not land on it, the MCP management part doesn't really matter in terms of actual functionality. I don't have a strong opinion either way though